eclipse-theia/theia
# support persistent websocket

Signed-off-by: bob <bob170731@gmail.com>



#### What it does


Support persistent websocket connection, which allow network to be weak and will not fire close event for shortly disconnected.


#### How to test


just open an ide in browser and fake network weakness by turning the computer with network-off mode, then turn on the network and the ide should continue to work fine.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
